xbmc/video/VideoDatabase.cpp

```diff
@@ -3950,44 +3939,55 @@ void CVideoDatabase::GetSameVideoItems(CFileItem& item, CFileItemList& items)
   {
     // get items with same unique ids
     m_pDS->query(
-        PrepareSQL("SELECT media_id "
+        PrepareSQL("SELECT DISTINCT media_id "
                    "FROM uniqueid "
                    "WHERE (value, type) IN "
                    "  (SELECT value, type FROM uniqueid WHERE media_id = %i AND media_type = '%s')",
                    dbId, mediaType.c_str()));
 
     while (!m_pDS->eof())
     {
-      insertItemId(m_pDS->fv("media_id").get_asInt());
+      itemIds.push_back(m_pDS->fv("media_id").get_asInt());
       m_pDS->next();
     }
 
     m_pDS->close();
 
-    // get items with same title
-    std::string title = item.GetVideoInfoTag()->GetTitle();
-
     VideoDbContentType itemType = item.GetVideoContentType();
+
+    // get items with same title and year
     if (itemType == VideoDbContentType::MOVIES)
     {
-      m_pDS->query(PrepareSQL("SELECT idMovie FROM movie WHERE movie.c%02d = '%s'",
-                              VIDEODB_ID_TITLE, title.c_str()));
+      if (item.GetVideoInfoTag()->HasYear())
+        m_pDS->query(PrepareSQL("SELECT idMovie FROM movie WHERE movie.c%02d = '%s' AND "
+                                "movie.premiered LIKE '%i%%'",
+                                VIDEODB_ID_TITLE, item.GetVideoInfoTag()->GetTitle().c_str(),
+                                item.GetVideoInfoTag()->GetYear()));
+      else
+        m_pDS->query(
+            PrepareSQL("SELECT idMovie FROM movie WHERE movie.c%02d = '%s' AND NOT movie.premiered",
```

Comment:
I'm not convinced yet that `movie.premiered LIKE '%i%%'` is best but let's go with that for now, it's used in many places already.
movie.premiered is text not a bool, I think testing movie.premiered = '0' is more correct.

Reply:
@CrystalP I tested the premiered column, seems an empty string works too, is "0" the default value for this column? what about using "LENGTH(premiered) < 4" as a correct premiered column length will be at least 4?

Reply:
premiered is text. no default that I can see in code.

The field is serialized by `CVideoDatabase::SetDetailsForMovie()` / `UpdateDetailsForMovie()` with
```C++
    if (details.HasPremiered())
      sql += PrepareSQL(", premiered = '%s'", details.GetPremiered().GetAsDBDate().c_str());
    else
      sql += PrepareSQL(", premiered = '%i'", details.GetYear());
```
Follow the code, looks to me like any null, blank or negative values of premiered would be bugs.
0 is the value that means "unknown". I only found rows with 0, YYYY or YYYY-MM-DD in my db.
In code, `GetYear()` returns 0 when invalid instead of the -1 it may contain internally. Same with premiered, returns 0 when invalid. All the pieces fit for me.

Comparison with empty string doesn't work here with sqlite, as expected because '' is not the same string as '0'

Reply:
This is how the column is interpreted in code:

CVideoDatabase::GetDetailsForMovie:

  if (premieredString.size() == 4)
    details.SetYear(record->at(VIDEODB_DETAILS_MOVIE_PREMIERED).get_asInt());
  else
    details.SetPremieredFromDBDate(premieredString);

where the SetPremieredFromDBDate method will use CDateTime::SetFromDBDate to interpret the value

bool CDateTime::SetFromDBDate(const std::string &date)
{
  if (date.size() < 10)
    return false;

So the length of this column is at least 4, less than that will be invalid. I can't find anywhere refer the "0" string. If there is no concern from you, I'll go with the "LENGTH(premiered) < 4" way.

Reply:
Fine. It's getting silly and for this simple case many tests work.

You don't find the '0' in code because, it's the 0 integer value returned by `GetYear()` for invalid dates, converted to text.